src/util/InstallDriver.ts

```diff
@@ -495,10 +498,48 @@ class InstallDriver {
     this.mApi.dismissNotification(getUnfulfilledNotificationId(collection.id));
     this.mApi.store.dispatch(actions.setModEnabled(profile.id, collection.id, true));
 
-    const required = (collection?.rules ?? [])
+    const required = []
       .filter(rule => rule.type === 'requires');
-    this.mRequiredMods = required
-      .filter(rule => util.findModByRef(rule.reference, mods) === undefined);
+    for (const rule of (collection?.rules ?? [])) {
+      if (rule.type !== 'requires') {
+        continue;
+      }
+
+      const modByRef = util.findModByRef(rule.reference, mods);
+      if (modByRef === undefined) {
+        required.push(rule);
+        continue;
+      }
+
+      if (rule.fileList?.length ?? 0 > 0) {
+        const stagingPath = selectors.installPathForGame(state, profile.gameId);
+        const modPath = path.join(stagingPath, modByRef.installationPath);
+        const presentFiles = await new Promise(async (resolve, reject) => {
```

Comment:
Why do you need this "await new Promise" here? It seems to me like you could just run the code in this block directly.